src/reducers/entities/search.js

```diff
@@ -61,12 +61,122 @@ function recent(state = {}, action) {
     }
 }
 
+/*
+ * Search reducer for mattermost-webapp will replace `SearchStore`
+ * Structure is:
+ * {
+ *     term: string on null - searching terms,
+ *     searchResults: array or null - list of searched posts
+ *     isMentionSearch: boolean - is searched be mentions
+ *     isFlaggedPosts: boolean - is searched flagged posts
+ *     isPinnedPosts: boolean - is searched boolean posts
+ * }
+ */
+function search(
```

Comment:
Getting there but we should split out the flagged posts and pinned posts into their own reducers and instead of adding a new search reducer, we'd have:

* `results` - This already exists and we should re-use it for storing the results of a regular search or a mention search. No need to differentiate between the two here since they're pretty much the same. We can track the difference in the webapp reducer if we need to
* `flaggedPosts` - This will be new and just store the post ids of the flagged posts, the actual posts should be stored in the `posts` reducer. E.g. `Set[postid1, postid2, etc...]`
* `pinnedPosts` - This will be new and store the pinned post ids for each channel E.g. `{channelid1: Set[postid1, postid2, etc...], channelid2: Set[], etc...}`

The `isMentionSearch`, `isFlaggedPosts`, `isPinnedPosts` and `term` fields don't need to be tracked in mattermost-redux because they're all controlling view state. The reducer that gets added to the webapp should take care of that view state (preferably using an enum var instead of individual booleans). The idea here is that post searches, flagged posts and pinned posts aren't really related so we shouldn't make them so in the reducers. We should be able to store them all in parallel without affecting the other and then the client (the webapp in our case) can choose how to show them

Reply:
@jwilander Thank you for your feedback. Will update PR soon